### PR TITLE
Test/unit tests

### PR DESCRIPTION
**Pull request by SarrankanPathmanathan**
Unit tests for all classes except for controllers since the testing the controllers will be part of Integration Testing.

### CODE REVIEW

File: src/main/java/tests/UnitTests/TestCommentPostRepository.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void testCommentPost() throws Exception {
+        assertNotNull(commentPost);
+        assertNotNull(submissionPost);
+    }
+
+    @Test
+    public void testFindPostByTitle() throws Exception{
```
Comment (stephanernst) [inline src/main/java/tests/UnitTests/TestCommentPostRepository.java]: Shouldn't this be called testFindPostByContext() ?

File: src/main/java/tests/UnitTests/TestSubmissionPost.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package tests.UnitTests;
+
+import ama.account.User;
+import ama.post.SubmissionPost;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class TestSubmissionPost {
```
Comment (stephanernst) [inline src/main/java/tests/UnitTests/TestSubmissionPost.java]: Should add a test for same user liking a post twice.

Comment (SarrankanPathmanathan) [inline src/main/java/tests/UnitTests/TestSubmissionPost.java]: Agreed and added

File: src/main/java/ama/WebSecurityConfiguration.java [hunk 3]
```diff
@@ -31,7 +31,6 @@ public BCryptPasswordEncoder bCryptPasswordEncoder() {
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
-                .csrf().disable()
```
Comment (AboudS777) [inline src/main/java/ama/WebSecurityConfiguration.java]: Why was this removed?

Comment (SarrankanPathmanathan) [inline src/main/java/ama/WebSecurityConfiguration.java]: We had to disable since we couldnt pass csrf tokens in our previous tests. Now that the tests use Mockmvc we can also pass the token within the request. So we dont need to disable.

Comment (SarrankanPathmanathan) [inline src/main/java/ama/WebSecurityConfiguration.java]: It wasnt that ur tests were useless. It was already being covered in the integration tests. So it became redundant

File: src/main/java/tests/UnitTests/TestCommentPost.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        kan = null;
+        submissionPost = null;
+        commentPost = null;
+    }
+
+    @Test
+    public void testCommentPost() throws Exception {
+        assertNotNull(sarran);
+        assertNotNull(submissionPost);
+        assertNotNull(commentPost);
```
Comment (RyanBillard) [inline src/main/java/tests/UnitTests/TestCommentPost.java]: this shouldn't be a test. You know none of these are null

File: src/main/java/tests/UnitTests/TestCommentPostRepository.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @After
+    public void tearDown() throws Exception {
+        commentPost = null;
+        submissionPost = null;
+    }
+
+    @Test
+    public void testCommentPost() throws Exception {
+        assertNotNull(commentPost);
+        assertNotNull(submissionPost);
```
Comment (RyanBillard) [inline src/main/java/tests/UnitTests/TestCommentPostRepository.java]: same thing. Don't need these null tests

File: src/main/java/tests/UnitTests/TestUserService.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    public void testRegisterNewUserAccount() throws Exception {
+        User registeredUser = userService.registerNewUserAccount(sarran);
+        assertEquals(registeredUser.getUsername(),sarran.getUsername());
+        assertEquals(registeredUser.getPassword(),sarran.getPassword());
+    }
+
+    @Test(expected= NullPointerException.class)
+    public void testLoadUserThatDoesNotExistByUsername() throws Exception {
+        UserDetails loadedUser = userService.loadUserByUsername(ryan.getUsername());
+        assertEquals(loadedUser.getUsername(),ryan.getUsername());
```
Comment (RyanBillard) [inline src/main/java/tests/UnitTests/TestUserService.java]: can't you just assert that loadedUser == null?

File: src/main/java/ama/account/UserService.java [hunk 7]
```diff
@@ -24,7 +24,7 @@ public User registerNewUserAccount(User user){
     }
 
     @Override
-    public User loadUserByUsername(String username) throws UsernameNotFoundException {
+    public User loadUserByUsername(String username) throws NullPointerException {
```
Comment (stephanernst) [inline src/main/java/ama/account/UserService.java]: I believe it was changed because userRepository.findByUsername() never actually throws a UsernameNotFoundException it simply returns null.

File: src/main/java/tests/UnitTests/TestCommentPost.java (hunk 4 above)
Comment (SarrankanPathmanathan) [inline src/main/java/tests/UnitTests/TestCommentPost.java]: Agreed, have been removed

File: src/main/java/tests/UnitTests/TestCommentPostRepository.java (hunk 5 above)
Comment (SarrankanPathmanathan) [inline src/main/java/tests/UnitTests/TestCommentPostRepository.java]: Agreed and removed

File: src/main/java/tests/UnitTests/TestUserService.java (hunk 6 above)
Comment (SarrankanPathmanathan) [inline src/main/java/tests/UnitTests/TestUserService.java]: True, Done.
